docs/guides/single-fetch.md

```diff
@@ -4,99 +4,95 @@ title: Single Fetch
-
-## Details
-
-### Streaming Data Format
+**1. Enable the future flag**
 
-Previously, Remix used `JSON.stringify` to serialize your loader/action data over the wire, and needed to implement a custom streaming format to support `defer` responses.
+```diff
+  remix({
+    future: {
+      // ...
++     unstable_singleFetch: true,
+    },
+  }),
+```
 
-With Single Fetch, Remix now uses [`turbo-stream`][turbo-stream] under the hood which provides first class support for streaming and allows you to automatically serialize/deserialize more complex data than JSON. The following data types can be streamed down directly via `turbo-stream`: `BigInt`, `Date`, `Error`, `Map`, `Promise`, `RegExp`, `Set`, `Symbol`, and `URL`. Subtypes of `Error` are also supported as long as they have a globally available constructor on the client (`SyntaxError`, `TypeError`, etc.).
+**2. Update or remove `installGlobals`**
 
-This may or may not require any immediate changes to your code once enabling Single Fetch:
+Single Fetch requires using [`undici`][undici] as your `fetch` polyfill, or using the built-in `fetch` on Node 20+, because it relies on APIs available there that are not in the `@remix-run/web-fetch` polyfill. Please refer to the [Undici][undici-polyfill] section in the 2.9.0 release notes below for more details.
 
-- ✅ `json` responses returned from `loader`/`action` functions will still be serialized via `JSON.stringify` so if you return a `Date`, you'll receive a `string` from `useLoaderData`/`useActionData`
-- ⚠️ If you're returning a `defer` instance or a naked object, it will now be serialized via `turbo-stream`, so if you return a `Date`, you'll receive a `Date` from `useLoaderData`/`useActionData`
-  - If you wish to maintain current behavior (excluding streaming `defer` responses), you may just wrap any existing naked object returns in `json`
+- If you are using Node 20+, remove `installGlobals()` and use Node's built-in `fetch` (this is the same thing as `undici`).
+- If you are managing your own server and calling `installGlobals()`, you will need to call `installGlobals({ nativeFetch: true })` to use `undici`.
 
-This also means that you no longer need to use the `defer` utility to send `Promise` instances over the wire! You can include a `Promise` anywhere in a naked object and pick it up on `useLoaderData().whatever`. You can also nest `Promise`'s if needed - but beware of potential UX implications.
+  ```diff
+  - installGlobals()
+  + installGlobals({ nativeFetch: true })
+  ```
 
-Once adopting Single Fetch, it is recommended that you incrementally remove the usage of `json`/`defer` throughout your application in favor of returning raw objects.
+- If you are using `remix-serve`, it will use `undici` automatically if Single Fetch is enabled.
 
-### React Rendering APIs
+**3. Replace `renderToString`**
```

Comment:
Good idea, I added this to the top

> For most Remix apps it's unlikely you're using `renderToString`, but if you have opted into using it in your `entry.server.tsx`, then continue reading, otherwise you can skip this step.

Reply:
ps. when I reading this, I am reminded of the `Inline Scripts` section below.

Both may not apply to everyone, but your app would break if you use these things and did not update.

Maybe both of those things should go into this step together.